packages/consensus/finality/finality_gadget.go

```diff
@@ -172,6 +176,29 @@ func (s *SimpleFinalityGadget) IsMessageConfirmed(msgID tangle.MessageID) (confi
 	return
 }
 
+// FirstUnconfirmedMarkerIndex returns the first Index in the given Sequence that was not confirmed, yet.
+func (s *SimpleFinalityGadget) FirstUnconfirmedMarkerIndex(sequenceID markers.SequenceID) (index markers.Index) {
+	s.lastConfirmedMarkersMutex.Lock()
+	defer s.lastConfirmedMarkersMutex.Unlock()
+
+	// TODO: MAP GROWS INDEFINITELY
```

Comment:
If we merge with this unsolved, we should open an issue and change the comment to "ISSUE XXXX: MAP GROWS INDEFINITELY"

I just don't think bugs should be tracked with TODOs

Reply:
Yep, totally right; Create issue https://github.com/iotaledger/goshimmer/issues/2016